File: app/api/run-scraper/route.ts
import { NextResponse } from 'next/server'
import path from 'path'
import fs from 'fs'

export async function GET(request: Request) {
  try {
    console.log('Starting scraper...')
    
    // Call the Python serverless function
    const response = await fetch(`${request.url.replace('/api/run-scraper', '/api/scrape')}`)
    const data = await response.json()
    
    if (!response.ok) {
      console.error('Error running scraper:', data.error)
      return NextResponse.json(
        { error: data.error },
        { status: response.status }
      )
    }
    
    console.log('Scraper completed successfully')
    
    // Check if the reports directory exists and list the files
    const reportsDir = path.join(process.cwd(), 'public', 'reports')
    console.log('Reports directory:', reportsDir)
    
    if (fs.existsSync(reportsDir)) {
      const files = fs.readdirSync(reportsDir)
        .filter(file => file.endsWith('.json') && file !== 'available-reports.json')
        .sort((a, b) => b.localeCompare(a)) // Sort in descending order (newest first)
      
      console.log('Reports found:', files)
      
      return NextResponse.json({
        success: true,
        message: 'Scraper completed successfully',
        reports: files
      })
    } else {
      console.error('Reports directory not found:', reportsDir)
      return NextResponse.json(
        { error: 'Reports directory not found' },
        { status: 500 }
      )
    }
  } catch (error) {
    console.error('Error running scraper:', error)
    return NextResponse.json(
      { error: error instanceof Error ? error.message : 'Unknown error' },
      { status: 500 }
    )
  }
} 

File: app/api/scrape/route.py
from http.server import BaseHTTPRequestHandler
import json
import os
import sys
import subprocess
from pathlib import Path

# Add the scripts directory to the Python path
scripts_dir = Path(__file__).parent.parent.parent.parent / 'scripts'
sys.path.append(str(scripts_dir))

# Import the scraper module
from tech_business_scraper import main as run_scraper

class handler(BaseHTTPRequestHandler):
    def do_GET(self):
        try:
            # Run the scraper
            run_scraper()
            
            # Get the list of reports
            reports_dir = Path(__file__).parent.parent.parent.parent / 'public' / 'reports'
            reports = [f for f in os.listdir(reports_dir) if f.endswith('.json') and f != 'available-reports.json']
            reports.sort(reverse=True)  # Sort in descending order (newest first)
            
            # Return the list of reports
            self.send_response(200)
            self.send_header('Content-type', 'application/json')
            self.end_headers()
            self.wfile.write(json.dumps({
                'success': True,
                'message': 'Scraper completed successfully',
                'reports': reports
            }).encode())
        except Exception as e:
            # Return the error
            self.send_response(500)
            self.send_header('Content-type', 'application/json')
            self.end_headers()
            self.wfile.write(json.dumps({
                'error': str(e)
            }).encode()) 